Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -80,18 +100,42 @@ TEST_F(SerializerTest, EncodesNullProtoToBytes) {
   // sanity check (the _init_default above should set this to _NULL_VALUE)
   EXPECT_EQ(google_protobuf_NullValue_NULL_VALUE, proto.value.null_value);
 
-  /* NB: proto bytes were created via:
-       echo 'null_value: NULL_VALUE' \
-         | ./build/external/protobuf/src/protobuf-build/src/protoc \
-             -I./Firestore/Protos/protos \
-             -I./build/external/protobuf/src/protobuf/src/ \
-             --encode=google.firestore.v1beta1.Value \
-             google/firestore/v1beta1/document.proto \
-             > output.bin
-   */
+  // TEXT_FORMAT_PROTO: 'null_value: NULL_VALUE'
   std::vector<uint8_t> bytes{0x58, 0x00};
   ExpectRoundTrip(proto, bytes, FieldValue::Type::Null);
 }
 
+TEST_F(SerializerTest, EncodesBoolModelToProto) {
+  for (bool test : {true, false}) {
+    FieldValue model = FieldValue::BooleanValue(test);
+    Serializer::TypedValue proto{FieldValue::Type::Boolean,
+                                 google_firestore_v1beta1_Value_init_default};
+    proto.value.boolean_value = test;
+    ExpectRoundTrip(model, proto, FieldValue::Type::Boolean);
+  }
+}
+
+TEST_F(SerializerTest, EncodesBoolProtoToBytes) {
+  struct TestCase {
+    bool value;
+    std::vector<uint8_t> bytes;
+  };
+
+  // TEXT_FORMAT_PROTO: 'boolean_value: true'
+  std::vector<uint8_t> true_bytes{0x08, 0x01};
+  // TEXT_FORMAT_PROTO: 'boolean_value: false'
+  std::vector<uint8_t> false_bytes{0x08, 0x00};
+
+  for (const TestCase& test :
+       {TestCase{true, true_bytes}, TestCase{false, false_bytes}}) {
```

Comment:
This is a little clumsy specified inline in the loop this way. Also it needlessly separates the boolean from the bytes in the input. Maybe initialize the TestCases above?

```c++
// TEXT_FORMAT_PROTO: 'boolean_value: true'
TestCase true_case{true, {0x08, 0x01}};

// TEXT_FORMAT_PROTO: 'boolean_value: false'
TestCase false_case{false, {0x08, 0x00}};

for (const TestCase& test : {true_case, false_case}) {
  // ...
}
```

Or even:
```c++
std::vector<TestCase> cases{
  // TEXT_FORMAT_PROTO: 'boolean_value: true'
  {true, {0x08, 0x01}},

  // TEXT_FORMAT_PROTO: 'boolean_value: false'
  {false, {0x08, 0x00}
};

for (const TestCase& test : cases) {
  // ...
}
```

Reply:
I really like the latter option; done. (The style formatter really does a number on it though.)